### PR TITLE
AC: add mixed adapter and regression adapter

### PR DESCRIPTION
**Pull request by tlsdmstn56-2**
* `RegressionAdapter`: outputs `RegressionPrediction`
* `MixedAdapter`: outputs `ContainerPrediction`. It can use with multiple adapters specified in config file as following:

```yml
adapter:
  type: mixed
  adapters:
  - type: classification
    layer_map:
      prob/gender: gender
      prob/hat: hat
  - type: regression
    layer_map:
      fc/age_mean/BiasAdd/Add: age
  - type: landmarks_regression
    layer_map:
      fc/fld/BiasAdd/Add: facial_landmark
```

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins ok to test

**Comment by eaidova** [comment]
@tlsdmstn56-2 in some cases it will not work or will duplicate info (e.g. if adapter required some additional parameters or several layers for processing).

On my view better reformat  adapters like:
```
adapters:
  <key_in_representation>: <adapter_configuration>
```
adapter configuration can be represented string or dict,
you do not need divide outputs, because each adapter works with its own layers

e.g.
```yml
adapter:
  type: mixed
  adapters:
    gender: 
      type: classification
        output_blob: prob/gender
    hat:
      type: classification
        output_blob: prob/hat
   age:
     type: regression
       output_blob:   fc/age_mean/BiasAdd/Add
   facial_landmark:
      type: landmarks_regression
        output_blob:  fc/fld/BiasAdd/Add
```

File: tools/accuracy_checker/accuracy_checker/adapters/README.md [hunk 1]
```diff
@@ -238,4 +238,7 @@ AccuracyChecker supports following set of adapters:
 * `face_detection_refinement` - converts output of face detection refinement model to `DetectionPrediction` representation. Adapter refines candidates generated in previous stage model.
     * `threshold` - Score threshold to determine as valid face candidate.
 * `attribute_classification` - converts output of attributes classifcation model to `ContainerPrediction` which contains multiple `ClassificationPrediction` for attributes with their scores.
-    * `output_layer_map` - dictionary where keys are output layer names of attribute classification model and values are the names of attributes.
\ No newline at end of file
+    * `output_layer_map` - dictionary where keys are output layer names of attribute classification model and values are the names of attributes.
+* `regression` - converting output of classification model to `RegressionPrediction` representation.
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/README.md:242]: classification model? not regression?

Comment (tlsdmstn56-2) [reply tools/accuracy_checker/accuracy_checker/adapters/README.md]: It should be regression. I will fix it in the next commit.

**Comment by tlsdmstn56-2** [comment]
In your format, why `output_blob` is not in the same depth as `type`? I think the following format looks better for me.

```yml
adapter:
  type: mixed
  adapters:
    gender: 
      type: classification
      output_blob: prob/gender
    hat:
      type: classification
      output_blob: prob/hat
   age:
     type: regression
     output_blob:   fc/age_mean/BiasAdd/Add
   facial_landmark:
     type: landmarks_regression
     output_blob:  fc/fld/BiasAdd/Add
```

**Comment by eaidova** [comment]
> In your format, why `output_blob` is not in the same depth as `type`? I think the following format looks better for me.

I am sorry, It is just problem with formatting, yes  I mean the same format like your said

File: tools/accuracy_checker/accuracy_checker/adapters/mixed_adapter.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+        for layer, (_, adapter) in self.adapters.items():
+            result[layer] = adapter.process(raw, identifiers, frame_meta)
+
+        if not self.is_result_valid(result):
+            raise RuntimeError("length of predictions from each adapter should be same")
+
+        output = []
+
+        for i in range(len(identifiers)):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mixed_adapter.py:81]:
```suggestion
        for i , _ in enumerate(identifiers):
```

File: tools/accuracy_checker/accuracy_checker/adapters/mixed_adapter.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+        for layer, (_, adapter) in self.adapters.items():
+            result[layer] = adapter.process(raw, identifiers, frame_meta)
+
+        if not self.is_result_valid(result):
+            raise RuntimeError("length of predictions from each adapter should be same")
+
+        output = []
+
+        for i , _ in enumerate(identifiers):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/mixed_adapter.py:81]:
```suggestion
        for i, _ in enumerate(identifiers):
```

**Pull request merged**